multiview/embed/gcca.py

```diff
@@ -0,0 +1,251 @@
+            default=False Set to true if n_samples > n_features, speeds up SVD
+
+        Attributes
+        ----------
+        _projection_mats : list of arrays A projection matrix for each view,
+            from the given space to the latent space self._ranks : list of ints
+            number of left singular vectors kept for each view during the first
+            SVD
+        """
+
+        Xs = check_Xs(Xs, multiview=True)
+        n = Xs[0].shape[0]
+        min_m = min(X.shape[1] for X in Xs)
+
+        self._check_inputs(
+            fraction_var=fraction_var,
+            sv_tolerance=sv_tolerance,
+            n_components=n_components,
+            size=(n, min_m),
+        )
+
+        data = [self._preprocess(x) for x in Xs]
+
+        Uall = []
+        Sall = []
+        Vall = []
+        ranks = []
+
+        for x in data:
+            # Preprocess
+            x[np.isnan(x)] = 0
+
+            # compute the SVD of the data
+            if tall:
+                v, s, ut = linalg.svd(x.T, full_matrices=False)
+            else:
+                u, s, vt = linalg.svd(x, full_matrices=False)
+                ut = u.T
+                v = vt.T
+
+            Sall.append(s)
+            Vall.append(v)
+            # Dimensions to reduce to
+            if sv_tolerance:
+                rank = sum(s > sv_tolerance)
+            elif n_components:
+                rank = n_components
+            else:
+                s2 = np.square(s)
+                rank = sum(np.cumsum(s2 / sum(s2)) < fraction_var) + 1
+            ranks.append(rank)
+
+            u = ut.T[:, :rank]
+            Uall.append(u)
+
+        d = min(ranks)
+
+        # Create a concatenated view of Us
+        Uall_c = np.concatenate(Uall, axis=1)
+
+        _, _, VV = svds(Uall_c, d)
+        VV = np.flip(VV.T, axis=1)
+        VV = VV[:, : min([d, VV.shape[1]])]
+
+        # SVDS the concatenated Us
+        idx_end = 0
+        projXs = []
+        projection_mats = []
+        for i in range(len(data)):
+            idx_start = idx_end
+            idx_end = idx_start + ranks[i]
+            VVi = normalize(VV[idx_start:idx_end, :], "l2", axis=0)
+
+            # Compute the canonical projections
+            A = np.sqrt(n - 1) * Vall[i][:, : ranks[i]]
+            A = A @ (linalg.solve(np.diag(Sall[i][: ranks[i]]), VVi))
+            projXs.append(data[i] @ A)
+            projection_mats.append(A)
+
+        self._projection_mats = projection_mats
+        self._ranks = ranks
+
+        return self
+
+    def transform(self, Xs, view_idx=None):
+        """
+        Embeds data matrix(s) using the fitted projection matrices
+
+        Parameters
+        ----------
+        Xs: list of array-likes - Xs shape: (n_views,) - Xs[i] shape:
+            (n_samples, n_features_i) The data to embed based on the prior fit
+            function view_idx: int The index of the view whose projection to
```

Comment:
description of view_idx is not clear to me. Is the purpose of it when Xs is a single view and you need to specify the projection matrix? If so, can you rewrite it so it's more clear.